### ISSUE TITLE
questions about supporting sets and query sets

### ISSUE THREAD

**Issue by OceanTangWei** [icoz69/DeepEMD#16]
Thanks for your nice work. I am new to FSL, so I am a little confused about supporting and query sets. In my understanding, we can only fine-tune the model on supporting sets, but I find you also use query set(each class 15), then what’s the difference between the case and 5way 20shot? I am really desperate for the answer. Thanks.

**Comment by icoz69** [comment]
hello
it is correct that only support set can be used for finetune the model during **the test phase**.  During meta-training phase, support set and query set are both used for optimization of the network.  In the test phase, the query set is only used for evaluation.

**Comment by OceanTangWei** [comment]
Thanks a lot. Here I have an another question, I find that when shot > 1, you directly calculate the mean value for the same class and treat those as the features for supporting sets,  won't it be too rough? Is there any better ways to improve the case where shot>1.

In Network.py:
 def get_sfc(self, support):
        support = support.squeeze(0)
        # init the proto

        SFC = support.view(self.args.shot, -1, 640, support.shape[-2], support.shape[-1]).mean(dim=0).clone().detach()

**Comment by icoz69** [comment]
hi this is only a step to initialize the SFC.  Later this SFC will further be finetuned by support set.
